fh-android-sdk/src/main/java/org/json/fh/CDL.java

```diff
@@ -191,31 +203,32 @@ public static JSONArray toJSONArray(JSONArray names, JSONTokener x)
     }
 
     /**
-     * Produce a comma delimited text row from a JSONArray. Values containing
+     * Produces a comma delimited text row from a JSONArray. Values containing
      * the comma character will be quoted.
-     * 
+     *
      * @param ja A JSONArray of strings.
      * @return A string ending in NEWLINE.
      */
     public static String rowToString(JSONArray ja) {
-        StringBuffer sb = new StringBuffer();
-        for (int i = 0; i < ja.length(); i += 1) {
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0, arrayLength = ja.length(); i < arrayLength; i++) {
```

Comment:
Prevents exposing `arrayLength` past the scope of the loop, but if you think it's messy/ugly, I can separate it.